File: Roslab2/Task1/second_pkg/second_pkg/node1.py
#!/usr/bin/env python3 
import  rclpy
from rclpy.node import Node
from example_interfaces.msg import UInt64 

class my_node(Node):
    def __init__(self):
        super().__init__("node1")
        self.counter=3
        self.get_logger().info("Node1 is Started")
        self.create_timer(2,self.timer_call_pub)
        self.obj_pub=self.create_publisher(UInt64,"uint64_topic",10)
        
	
    def timer_call_pub(self):
        msg_uint64=UInt64()
        msg_uint64.data= self.counter
        self.obj_pub.publish(msg_uint64)
        self.get_logger().info(str(msg_uint64.data))


def main (args=None):
    rclpy.init(args=args)
    node=my_node()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__=="__main__":
    main()

File: Roslab2/Task3/cpp_first_pkg/src/node1.cpp
#include <chrono>
#include<string>
#include <iostream>
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
 
class Node1 : public rclcpp::Node
{
public:
    Node1():Node("string_publisher")
    {
        RCLCPP_INFO(this->get_logger(),"string_publisher has started");
        string_publisher_ = this->create_publisher<std_msgs::msg::String>("str_topic",10);
        string_timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&Node1::timer_cb_pub_one, this) );
        int_publisher_ = this->create_publisher<std_msgs::msg::String>("int_topic",10);
        int_timer_ = this->create_wall_timer(std::chrono::milliseconds(500), std::bind(&Node1::timer_cb_pub_two, this) );
        flag_subscriber_ = this->create_subscription<std_msgs::msg::String>("flag_topic",10, std::bind(&Node1::timer_cb_sub, this, std::placeholders::_1));
    }
private:
    void timer_cb_pub_one()
    {
        std_msgs::msg::String string_msg = std_msgs::msg::String();
        string_msg.data = "Rouzan is puplishing, "+std::to_string(Node1::counter);
        Node1::counter++;
        string_publisher_->publish(string_msg);
    }
    void timer_cb_pub_two()
    {
        std_msgs::msg::String int_msg = std_msgs::msg::String();
        int_msg.data = std::to_string(Node1::counter);
        int_publisher_->publish(int_msg);
    }
    void timer_cb_sub(const std_msgs::msg::String::SharedPtr flag_msg)
    {
        Node1::flag=*(flag_msg->data.c_str());
        if (Node1::flag=='1'){
           Node1::counter=0;
           Node1::flag='0';
        }
       
    }
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr string_publisher_;
    rclcpp::TimerBase::SharedPtr string_timer_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr int_publisher_;
    rclcpp::TimerBase::SharedPtr int_timer_;
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr flag_subscriber_;
    int counter =0;
    char flag ='0';
};

int main(int argc, char * argv[])
{
    rclcpp::init(argc,argv);
    rclcpp::spin(std::make_shared<Node1>());
    rclcpp::shutdown();
    return 0;
} 
